cranelift/codegen/src/isa/zkasm/lower.isle

```diff
@@ -31,15 +31,10 @@
 (rule 0 (lower (has_type (ty_int_ref_scalar_64 ty) (iadd x y)))
   (rv_add x y))
 
-;; TODO: restore it, disabled cause matches 1 as i64 but don't correspond our i32 
-;; representation.
+(rule 1 (lower (has_type (fits_in_32 (ty_int ty)) (iadd x y)))
```

Comment:
I believe for masking like this `fits_in_32` is the wrong check to make here. It will also trigger all the same for $I8 and $I16 types (yes, WASM hasn't those, but there's nothing stopping cranelift from generating them all the same.)

Should just match on `(has_type $I32)` here like is done in other places within this file.

(Longer term I believe we should not have the masking as present in CLIF as much. What I believe we should do is add an extern function that would generate the necessary masking depending on the type, so that you could write something like

```
(rule 0 (lower (iadd x y))
  (masked (zk_add x y))
)
```

or whatever.)

Reply:
Good point, replaced.